### PR TITLE
fixes(#2329)

### PR DESCRIPTION
**Pull request by haverchuck**
*Issue #, if available:*
fixes(#2329 )

*Description of changes:*
Added ability to use SignOut and Greetings outside of Authenticator (React).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 1]
```diff
@@ -26,22 +26,60 @@ const logger = new Logger('Greetings');
...
+                    authState: 'signedIn',
+                    authData: payload.data
+                })
+            } else if (channel === 'auth' && payload.event === 'customSignOut' && (!this.props.authState)) {
+                this.setState({
+                    authState: 'signIn'
+                })
+            }
+    
+            if (channel === 'auth' && payload.event === 'signIn' && (!this.props.authState)) {
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/Greetings.jsx]: Can we move this into the block line 60 to 64

File: packages/aws-amplify-react/src/Auth/SignOut.jsx [hunk 2]
```diff
@@ -28,11 +28,52 @@ export default class SignOut extends AuthPiece {
         super(props);
 
         this.signOut = this.signOut.bind(this);
+        Hub.listen('auth', this)
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/SignOut.jsx]: Same as above.

File: packages/aws-amplify-react/src/Auth/SignOut.jsx [hunk 3]
```diff
@@ -28,11 +28,52 @@ export default class SignOut extends AuthPiece {
...
+        if (!this.props.authState && !this.props.authData) {
+            Auth.currentAuthenticatedUser()
+            .then(user => {
+                this.setState({
+                    authState: 'signedIn',
+                    authData: user,
+                    stateFromStorage: true
+                })
+            })
+            .catch(err => console.log(err));
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/SignOut.jsx]: Same as above.

File: packages/aws-amplify-react/src/Auth/SignOut.jsx [hunk 4]
```diff
@@ -28,11 +28,52 @@ export default class SignOut extends AuthPiece {
...
+                    authState: 'signedIn',
+                    authData: payload.data
+                })
+            } else if (channel === 'auth' && payload.event === 'customSignOut' && (!this.props.authState)) {
+                this.setState({
+                    authState: 'signIn'
+                })
+            }
+    
+            if (channel === 'auth' && payload.event === 'signIn' && (!this.props.authState)) {
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/SignOut.jsx]: Same as above.

Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/SignOut.jsx]: done.

File: packages/aws-amplify-react/src/Auth/SignOut.jsx (hunk 3 above)
Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/SignOut.jsx]: done.

File: packages/aws-amplify-react/src/Auth/SignOut.jsx (hunk 2 above)
Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/SignOut.jsx]: done

File: packages/aws-amplify-react/src/Auth/Greetings.jsx (hunk 1 above)
Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: done.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 5]
```diff
@@ -26,22 +26,60 @@ const logger = new Logger('Greetings');
...
+        if (!this.props.authState && !this.props.authData) {
+            Auth.currentAuthenticatedUser()
+            .then(user => {
+                this.setState({
+                    authState: 'signedIn',
+                    authData: user,
+                    stateFromStorage: true
+                })
+            })
+            .catch(err => console.log(err));
```
[the remark this answers is not in the document]
Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: done.

File: packages/aws-amplify-react/src/Auth/Greetings.jsx [hunk 6]
```diff
@@ -26,22 +26,60 @@ const logger = new Logger('Greetings');
 export default class Greetings extends AuthPiece {
     constructor(props) {
         super(props);
+        this.state = {};
+        Hub.listen('auth', this)
```
[the remark this answers is not in the document]
Comment (haverchuck) [reply packages/aws-amplify-react/src/Auth/Greetings.jsx]: done.

**Pull request merged**